torch/csrc/api/include/torch/functional.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <torch/types.h>
+
+namespace torch {
+
+inline at::Tensor isfinite(const at::Tensor& tensor) {
```

Comment:
We should move the implementation of python isfinite to c++ unless there is some reason why the python version cannot call into c++. This way, we'll have just one implementation for both: https://pytorch.org/docs/master/_modules/torch/functional.html#isfinite.

Reply:
Yes agreed that it would be awesome to unify these two.

Reply:
To clarify, to do this, one would:
- add a new `isfinite` entry into native_functions.yaml: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml
- delete the `torch/functional.py` version of `isfinite` (it won't be  necessary anymore)
- while preserving all documentation (the torch/functional.py isfinite docs might need to be moved to _torch_docs.py.